packages/components/src/Button/Button.tsx

```diff
@@ -53,7 +53,9 @@ interface ButtonAnchorProps extends ButtonFoundationProps {
 
 interface ButtonLinkProps<S = unknown> extends ButtonFoundationProps {
   /**
-   * Used for client side routing. Only use when inside a routed component.
+   * **Deprecated**: to will be removed in the next major version
+   * Use shared ButtonLink component from Jobber instead
+   * @deprecated
```

Comment:
Okay so the idea here is to deprecate the prop until we issue the next major version change? 

I think that might line up well with the ESM change I'm prepping now, so consumers might not have a ton of warning, but we could absolutely write a small documentation piece that suggests how to fix this (either use the ButtonLink in JO, or implement it yourselves in your own project).

My only suggestion about the copy/text above is recommending consumers `use shared ButtonLink component from Jobber instead`. Since Atlantis is an open-source repo and we do have consumers outside the JO ecosystem, we might need to remove that line, maybe opt for a link to that short bit of documentation we'll need to write?